### PR TITLE
fix: json data expection while converting reason type

### PR DESCRIPTION
**Pull request by cre8ivejp**
Due to changes made in this [PR](https://github.com/bucketeer-io/android-client-sdk/pull/63) in the API response, some users are getting the following error.

```
Caused by: com.squareup.moshi.JsonDataException: Expected a string but was NUMBER at path $.reason.type
``` 

Because the reason object has changed, we must delete the old evaluation data in the evaluation table while upgrading the SDK.

### CODE REVIEW

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/database/DBOpenHelper.kt [hunk 1]
```diff
@@ -1,49 +1,59 @@
...
 
 class OpenHelperCallback : SupportSQLiteOpenHelper.Callback(VERSION) {
 
   companion object {
     const val FILE_NAME = "bucketeer.db"
-    const val VERSION = 2
+    const val VERSION = 3
+
+    @VisibleForTesting
+    fun v2Schema(db: SupportSQLiteDatabase) {
```
Comment (cre8ivejp) [inline bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/database/DBOpenHelper.kt:20]: To avoid duplicate code, I put the schema in this static function so we can reuse it in the [tests](https://github.com/bucketeer-io/android-client-sdk/pull/79/files#diff-9e9eebfdac59b75de3adc119068791edf454ffff72a7cddfd427a804bbadb7a7R107).
If there is a better of doing this, please let me know.

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/database/migration/Migration2to3.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package io.bucketeer.sdk.android.internal.database.migration
+
+import androidx.sqlite.db.SupportSQLiteDatabase
+import io.bucketeer.sdk.android.internal.evaluation.db.EvaluationEntity
+
+class Migration2to3 : Migration {
+
+  override fun migrate(db: SupportSQLiteDatabase) {
+    db.delete(EvaluationEntity.TABLE_NAME, null, null)
```
Comment (duyhungtnn) [inline bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/database/migration/Migration2to3.kt:9]: Should we recreate `Evaluation` table again after delete ? @cre8ivejp

Comment (cre8ivejp) [reply bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/database/migration/Migration2to3.kt:9]: There is no need to recreate the table. I think the approach here is to delete the rows in the table or drop the table and create it again.
The delete operation would be fine since there is only one row in the table.

Also, I wrote a test to check if it is empty after migration.
https://github.com/bucketeer-io/android-client-sdk/pull/79/files#diff-9e9eebfdac59b75de3adc119068791edf454ffff72a7cddfd427a804bbadb7a7R86-R96

Comment (duyhungtnn) [reply bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/database/migration/Migration2to3.kt:9]: thank you, I checked it document.
> `delete` Convenience method for deleting rows in the database

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/database/migration/Migration2to3.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import android.content.SharedPreferences
+import androidx.sqlite.db.SupportSQLiteDatabase
+import io.bucketeer.sdk.android.internal.Constants
+import io.bucketeer.sdk.android.internal.evaluation.db.EvaluationEntity
+import io.bucketeer.sdk.android.internal.event.EventEntity
+
+class Migration2to3 : Migration {
+
+  override fun migrate(db: SupportSQLiteDatabase, sharedPreferences: SharedPreferences) {
+    db.delete(EvaluationEntity.TABLE_NAME, null, null)
```
Comment (yshrsmz) [inline bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/database/migration/Migration2to3.kt:12]: Can you add a comment that describes why we need to delete data in Evaluation table, so that we can better understand the intention in the future?

**Review by yshrsmz — DISMISSED** [review]
👍

**Review by yshrsmz — APPROVED** [review]
👍

File: bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/database/migration/Migration2to3.kt (hunk 3 above)
Comment (cre8ivejp) [reply bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/database/migration/Migration2to3.kt:15]: @yshrsmz I have added a comment.
https://github.com/bucketeer-io/android-client-sdk/pull/79/files#diff-439703daa4fc7dfb6c064763b3f3a622dd26367a1e749131e5195e8ab3e5c6f1R12-R14

Thanks!
